File: examSeatingAppLocal/UI/src/stores/examRooms.js
import { defineStore } from 'pinia'
import { ref, computed } from 'vue'
import { examRoomsApi } from '@/services/examRoomsApi'
import { useAppStore } from './app'

export const useExamRoomsStore = defineStore('examRooms', () => {
  // State
  const examRooms = ref([])
  const loading = ref(false)

  // Getters
  const roomCount = computed(() => examRooms.value.length)
  const totalCapacity = computed(() => 
    examRooms.value.reduce((total, room) => total + room.roomCapacity, 0)
  )

  // Actions
  const fetchExamRooms = async () => {
    try {
      loading.value = true
      const response = await examRoomsApi.getAll()
      // Handle both array and object responses
      examRooms.value = Array.isArray(response) ? response : (response.examRooms || [])
    } catch (error) {
      console.error('Failed to fetch exam rooms:', error)
      examRooms.value = []
    } finally {
      loading.value = false
    }
  }

  const createExamRoom = async (roomData) => {
    try {
      const appStore = useAppStore()
      await examRoomsApi.create(roomData)
      await fetchExamRooms() // Refresh the list
      appStore.showSuccess('Exam room created successfully')
    } catch (error) {
      console.error('Failed to create exam room:', error)
      throw error
    }
  }

  const updateExamRoom = async (roomId, roomData) => {
    try {
      const appStore = useAppStore()
      await examRoomsApi.update(roomId, roomData)
      await fetchExamRooms() // Refresh the list
      appStore.showSuccess('Exam room updated successfully')
    } catch (error) {
      console.error('Failed to update exam room:', error)
      throw error
    }
  }

  const deleteExamRoom = async (roomId) => {
    try {
      const appStore = useAppStore()
      await examRoomsApi.delete(roomId)
      await fetchExamRooms() // Refresh the list
      appStore.showSuccess('Exam room deleted successfully')
    } catch (error) {
      console.error('Failed to delete exam room:', error)
      throw error
    }
  }

  return {
    // State
    examRooms,
    loading,
    
    // Getters
    roomCount,
    totalCapacity,
    
    // Actions
    fetchExamRooms,
    createExamRoom,
    updateExamRoom,
    deleteExamRoom
  }
})

File: examSeatingAppLocal/fastapi_app/app/routes/exam_rooms.py
from fastapi import APIRouter
from typing import Dict
from ..models import ExamRoomModel
from ..services import ExamRoomService

router = APIRouter(prefix="/examRoom", tags=["exam_rooms"])

@router.get("", response_model=Dict)
def get_exam_rooms():
    """Get all exam rooms"""
    exam_rooms = ExamRoomService.get_all_exam_rooms()
    return {"examRooms": exam_rooms}

@router.post("")
def add_exam_room(room_data: ExamRoomModel):
    """Add a new exam room"""
    ExamRoomService.create_exam_room(room_data)
    return {"message": "Exam room added successfully"}

@router.put("/{room_id}")
def update_exam_room(room_id: int, room_data: ExamRoomModel):
    """Update an existing exam room"""
    ExamRoomService.update_exam_room(room_id, room_data)
    return {"message": "Exam room updated successfully"}

@router.delete("/{room_id}")
def delete_exam_room(room_id: int):
    """Delete an exam room"""
    ExamRoomService.delete_exam_room(room_id)
    return {"message": "Exam room deleted successfully"}
